Source/Core/DolphinQt2/DPI.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/DPI.h"
+
+#include <QGuiApplication>
+#include <QPaintDevice>
+#include <QPixmap>
+#include <QScreen>
+#include <QSize>
+#include <QString>
+#include <algorithm>
+
+double DPI::GetScale()
+{
+  const auto screen = QGuiApplication::primaryScreen()->geometry();
+
+  return (screen.width() + screen.height() / 2) /
+         (QGuiApplication::primaryScreen()->logicalDotsPerInch() * 18);
+}
+
+double DPI::GetScaledIconSize(const double scale)
+{
+  if (scale < 1.1)
+    return 1;
+  if (scale >= 1.1 && scale < 1.8)
+    return 2;
+
+  return 4;
+}
+
+const QString DPI::GetScaledIconSuffix(const double scale, const double max)
```

Comment:
Noticed this just now, but the const on the return type here (and in the prototype) can be dropped (returning by const value can inhibit move construction).